src/rust/scheduler/handle.rs

```diff
@@ -55,6 +77,38 @@ impl SchedulerHandle {
     }
 }
 
+impl YielderHandle {
+    pub fn new() -> Self {
+        Self {
+            result_handle: Rc::new(RefCell::new(None)),
+            waker_handle: Rc::new(RefCell::new(None)),
+        }
+    }
+
+    /// Wake this yielded coroutine: Ok indicates there is work to be done and Fail indicates the coroutine should exit
+    /// with an error.
+    pub fn wake_with(&mut self, result: Result<(), Fail>) {
+        if let Some(res) = self.result_handle.borrow_mut().replace(result) {
+            debug!("Wake already scheduled: overwriting previous wake result: {:?}", res);
```

Comment:
Could you prefix log messages with the function name? Otherwise, when running with logs enabled we cannot easily figure out where those messages come from. Also, consider adding the parameters that were passed to the logged function, to help on understanding the context.